picard/cluster.py

```diff
@@ -223,7 +221,7 @@ def column(self, column):
         elif self.special and column in {'~length', 'album', 'covercount'}:
             return ''
         elif column == '~length':
-            return format_time(self.metadata.length)
+            return format_time(sum(f.metadata.length for f in self.files))
```

Comment:
This might be called really a lot of times, and we might have huge clusters.

I think we should have a `length` property on `Cluster` that recalculates the value on demand and gets marked as dirty when files get added or removed.

Reply:
> @zas I like this approach of the list class.
> 
> Just the repeated calls to `self.set_dirty()` read a bit strange to me. Instead of a set I think it would be easier to read to have this a property and one can run `self.dirty = True`.
> 

Addressed in https://github.com/metabrainz/picard/pull/2684/commits/0f96974a145d874a0c62dfea70930c20d450163a

> Also I think we don#t need to mark it dirty on `sort` and `reverse`, as it doesn't change the element count.

True, removed in https://github.com/metabrainz/picard/pull/2684/commits/72e8fb0d8a03c7319e938ac5efd7111aaa41e6b3